experimental/framework/frs-client/src/FrsClient.ts

```diff
@@ -0,0 +1,163 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Container, Loader } from "@fluidframework/container-loader";
+import {
+    ContainerSchema,
+    DOProviderContainerRuntimeFactory,
+    FluidContainer,
+    RootDataObject,
+} from "@fluid-experimental/fluid-static";
+import {
+    IDocumentServiceFactory,
+} from "@fluidframework/driver-definitions";
+import { ITokenProvider, RouterliciousDocumentServiceFactory } from "@fluidframework/routerlicious-driver";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { InsecureTokenProvider } from "@fluidframework/test-runtime-utils";
+import { generateUser } from "@fluidframework/server-services-client";
+import {
+    FrsConnectionConfig,
+    FrsContainerConfig,
+    FrsContainerServices,
+} from "./interfaces";
+import { FrsAudience } from "./FrsAudience";
+import { FrsUrlResolver } from "./FrsUrlResolver";
+import { debug } from "./debug";
+
+/**
+ * FrsClientInstance provides the ability to have a Fluid object backed by the FRS service
+ */
+export class FrsClientInstance {
+    public readonly documentServiceFactory: IDocumentServiceFactory;
+    public readonly tokenProvider: ITokenProvider;
+    constructor(private readonly connectionConfig: FrsConnectionConfig) {
+        const user = this.connectionConfig.user
+            ? { id: this.connectionConfig.user.userId, name: this.connectionConfig.user.userName }
+            : generateUser();
+        this.tokenProvider = connectionConfig.tokenProvider
+            ?? new InsecureTokenProvider(connectionConfig.key, user);
+        this.documentServiceFactory = new RouterliciousDocumentServiceFactory(
+            this.tokenProvider,
+        );
+    }
+
+    public async createContainer(
+        containerConfig: FrsContainerConfig,
+        containerSchema: ContainerSchema,
+    ): Promise<[FluidContainer, FrsContainerServices]> {
```

Comment:
I'd prefer an object here, destructuring an object with named properties is less error-prone and easier typing, esp. if we make changes in the future.
```suggestion
    ): Promise<{ fluidContainer: FluidContainer, containerServices: FrsContainerServices }> {
```

Reply:
Hmm while I agree that it makes it easier for typing and making changes in the future, two points:
1) It was initially made a tuple intentionally to keep a pairing between the container and the services available to it. Creating an object here makes it easy to just add in properties to that object itself instead of trying to fit it into the container/containerService model
2) If we do want to switch it out from being a tuple, we should do it for all clients uniformly. Tinylicious/FRS/ODSP should all return a similar format so this might be a good follow-up to apply across the board if we choose to go down this route

Reply:
I kind of wonder if whoever did this originally intended it to be an object, considering it was written as `membername: type` (and broke vscode for me 👿)

Reply:
Agree on uniformly making the change.  I don't think using an array makes it any harder to expand it than an object though.